lib/matplotlib/legend.py

```diff
@@ -544,8 +544,11 @@ def __init__(self, parent, handles, labels,
             'mec':             ['get_markeredgecolor', 'get_edgecolor'],
         }
         if labelcolor is None:
-            pass
-        elif isinstance(labelcolor, str) and labelcolor in color_getters:
+            if mpl.rcParams['legend.labelcolor'] != 'none':
```

Comment:
The legend docstring should note that this rcparam exists....

Reply:
Should this be `
```suggestion
            if mpl.rcParams['legend.labelcolor'] is not None:
```

along with similar changes to validator?  There are a number of places where we use the string `'none'` as a color to mean "this thing does not _get_ a color" and `None` (the singleton) to mean "use the default color".

That said, currently doing

```
...
ax.legend(labelcolor='none')
```
does not result in an invisible label which is a bug in the implementation below (cycle of a 0 length thing is also 0 length, not infinite).

Reply:
The rcParam can't be able to both be `None` and 'none`, though, right? The fallback version would need a special value to work, like `'auto'` or `'textcolor'`?

Reply:
https://github.com/matplotlib/matplotlib/pull/20225 fixes the bug I noted above!